rpc/rpcutils/authcodec.go

```diff
@@ -89,55 +175,56 @@ func NewAuthServerCodec(conn io.ReadWriteCloser, codecToWrap rpc.ServerCodec, pa
 //  lets the underlying codec read the rest.
 //  Finally, it validates the identity if necessary.
 func (a *AuthServerCodec) ReadRequestHeader(r *rpc.Request) error {
+	// Lock so we read both values back-to-back
+	a.rBuffMutex.Lock()
+	defer a.rBuffMutex.Unlock()
```

Comment:
Mutex is not necessary here.  However a comment to the effect that you considered using a Mutex and determined against it would be useful.